src/Npgsql/PostgresVariables.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.IO;
+using Npgsql.Util;
+
+namespace Npgsql
+{
+    static class PostgresVariables
+    {
+        public static string? User => Environment.GetEnvironmentVariable("PGUSER") ?? Environment.UserName;
+
+        public static string? Password => Environment.GetEnvironmentVariable("PGPASSWORD");
+
+        public static string? PassFile => Environment.GetEnvironmentVariable("PGPASSFILE") ?? DefaultFilePath("pgpass.conf");
```

Comment:
Shouldn't this be `.pgpass` on Linux and `pgpass.conf` on Windows?

Reply:
While you're right, let's fix this in a different PR (: Openned #2684 to track this.